### PR TITLE
fix ktest to use JCommander

### PR DESCRIPTION
**Pull request by dwightguth**
The last of the refactorings to remove Apache Commons CLI.

@osa1 please review.

### CODE REVIEW

**Comment by dwightguth** [comment]
Broke the script that calls ktest on Jenkins because it still used the deprecated single hyphen.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

File: src/javasources/KTool/src/org/kframework/ktest/CmdArgs/KTestOptions.java [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+        this.ignoreWS = obj.ignoreWS;
+        this.ignoreBalancedParens = obj.ignoreBalancedParens;
+        this.dry = obj.dry;
+    }
+
+    /**
+     * Validate raw data parsed from command line arguments and return CmdArg objects,
+     * which is needed for TestSuite to run tests.
+     * @param cmdOpts CommandLine object generated by command line argument parser
+     * @return validated CmdArg object that is needed by TestSuite
```
Comment (osa1) [inline src/javasources/KTool/src/org/kframework/ktest/CmdArgs/KTestOptions.java]: `@param` and `@return` parts are not relevant anymore.

File: src/javasources/KTool/src/org/kframework/krun/api/Transition.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ public TransitionType getType() {
     public String toString() {
         if (type == TransitionType.RULE) {
             Attributes a = rule.getAttributes();
-            UnparserFilter unparser = new UnparserFilter(true, context.krunOptions.color(), context.krunOptions.output, context);
+            UnparserFilter unparser = new UnparserFilter(true, context.colorOptions.color(context.krunOptions.experimental.outputFile != null), context.krunOptions.output, context);
```
Comment (osa1) [inline src/javasources/KTool/src/org/kframework/krun/api/Transition.java]: I'm a bit confused about this part. We're using Jansi to handle output stream's color support. Do we still need to pass redirection info to color class?

Also, this doesn't work in general. Currently we're using `--output-file` argument but what happens if output is redirected using pipes etc. ? Jansi handles that too.

File: src/javasources/KTool/src/org/kframework/ktest/CmdArgs/CmdArg.java [hunk 3]
```diff
@@ -1,382 +0,0 @@
...
-
-    /**
-     * Dry run.
-     */
-    private final boolean dry;
-
-    /**
-     * Run in debugging mode. (prints stack traces when uncaught exceptions are thrown)
-     */
-    private final boolean debug;
```
Comment (osa1) [inline src/javasources/KTool/src/org/kframework/ktest/CmdArgs/CmdArg.java:119]: If I'm not missing anything, debug argument is missing in new version. That was being useful for getting stack traces. Why is it removed? Or is it moved to some other place?

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/ktest/CmdArgs/CmdArg.java:119]: debug is part of GlobalOptions now.

File: src/javasources/KTool/src/org/kframework/ktest/KTest.java [hunk 4]
```diff
@@ -85,59 +75,38 @@ private TestSuite makeTestSuite(String targetFile, CmdArg cmdArgs) throws SAXExc
...
+            }
+            KTest ktest = new KTest(options);
 
-        try {
-            return ktest.run() == 0;
+            return ktest.run();
         } catch (SAXException | ParserConfigurationException | IOException | InterruptedException |
-                TransformerException | InvalidArgumentException e) {
-            if (ktest.debug()) {
-                e.printStackTrace();
```
Comment (osa1) [inline src/javasources/KTool/src/org/kframework/ktest/KTest.java:132]: Maybe replace that part with `if (options.getGlobal().debug)) { e.printStackTrace(); }` ?

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/ktest/KTest.java:132]: see now

**Comment by osa1** [comment]
Thanks. LGTM.

**Pull request merged**